samples/apply-stretch-renderer/src/main/java/com/esri/arcgismaps/sample/applystretchrenderer/components/ApplyStretchRendererViewModel.kt

```diff
@@ -0,0 +1,177 @@
+
+    // Percent clip parameters (values represent percent 0..100)
+    private val _percentMin = MutableStateFlow(0.0)
+    val percentMin = _percentMin.asStateFlow()
+
+    private val _percentMax = MutableStateFlow(50.0)
+    val percentMax = _percentMax.asStateFlow()
+
+    // Standard deviation factor (typical range 0.25..4.0)
+    private val _stdDeviationFactor = MutableStateFlow(0.5)
+    val stdDeviationFactor = _stdDeviationFactor.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            try {
+                // Load the raster layer and apply an initial renderer
+                val raster = Raster.createWithPath(File(provisionPath, "raster-file${File.separator}Shasta.tif").path)
+                rasterLayer = RasterLayer(raster)
+                rasterLayer?.load()?.getOrThrow()
+
+                arcGISMap.apply {
+                    rasterLayer?.let {
+                        operationalLayers.add(it)
+                    }
+                }.load().getOrThrow()
+
+                val center = rasterLayer?.fullExtent?.center ?: throw Exception("Failed to get raster center")
+                mapViewProxy.setViewpoint(Viewpoint(center = center, scale = 80_000.0))
+                updateRenderer()
+            } catch (ex: Exception) {
+                messageDialogVM.showMessageDialog(ex)
+            }
+        }
+    }
+
+    /** Update the current stretch type by index from [stretchTypeOptions]. */
+    fun updateStretchTypeByIndex(index: Int) {
+        val safeIndex = index.coerceIn(0, stretchTypeOptions.lastIndex)
+        _selectedStretchType.value = when (safeIndex) {
+            0 -> StretchType.MinMax
+            1 -> StretchType.PercentClip
+            else -> StretchType.StandardDeviation
+        }
+    }
+
+    /** Update MinMax minimum value (clamped to 0..(max-1)). */
+    fun updateMinValue(value: Double) {
+        val max = _maxValue.value
+        _minValue.value = value.coerceIn(0.0, max - 1.0)
+    }
+
+    /** Update MinMax maximum value (clamped to (min+1)..255). */
+    fun updateMaxValue(value: Double) {
+        val min = _minValue.value
+        _maxValue.value = value.coerceIn(min + 1.0, 255.0)
+    }
+
+    /** Update Percent Clip minimum percent (clamped to 0..percentMax). */
+    fun updatePercentMin(value: Double) {
+        val max = _percentMax.value
+        _percentMin.value = value.coerceIn(0.0, max)
+    }
+
+    /** Update Percent Clip maximum percent (clamped to percentMin..100). */
+    fun updatePercentMax(value: Double) {
+        val min = _percentMin.value
+        _percentMax.value = value.coerceIn(min, 100.0)
+    }
+
+    /** Update Standard Deviation factor (clamped to 0.25..4.0). */
+    fun updateStdDeviationFactor(value: Double) {
+        _stdDeviationFactor.value = value.coerceIn(0.25, 4.0)
+    }
+
+    /** Construct and apply a StretchRenderer to the raster layer using current UI parameters. */
+    fun updateRenderer() {
```

Comment:
Nitpick - when I make changes to a renderer in the bottom sheet, then dismiss the bottom sheet, the changes are not applied to the map (as expected) but they persist in the bottom sheet and in the view model. The changes should only be applied to the view model when "update renderer is clicked".

Reply:
Made the updates 👍🏼